packages/core/src/components/chip/index.js

```diff
@@ -0,0 +1,78 @@
+class Chip {
+  static instances = new WeakMap();
+
+  static get cssClasses() {
+    return {
+      ACTIVE: 'ray-chip--active'
+    };
+  }
+
+  static get strings() {
+    return {
+      INIT_SELECTOR: '.ray-chip'
+    };
+  }
+
+  static create(element, options) {
+    return this.instances.get(element) || new this(element, options);
+  }
+
+  static createAll(target = document, _options = { active: false }) {
+    // Finds all instances of select on the document or within a given element and instantiates them.
+    const options = {
+      initSelector: this.strings.INIT_SELECTOR,
+      ..._options
+    };
+
+    const chips = Array.from(target.querySelectorAll(options.initSelector));
+    chips.forEach(select => this.create(select, options));
+  }
+
+  constructor(root, options) {
+    this._root = root;
+    this._options = options;
+    this.constructor.instances.set(this._root, this);
+
+    this.state = {
+      active: this._root.classList.contains(this.constructor.cssClasses.ACTIVE)
+    };
+
+    this._bindEventListeners();
+    this.assignClasses();
+  }
+
+  _bindEventListeners() {
+    this._root.addEventListener('mousedown', this.onClick);
+    this._root.addEventListener('keyup', this.onSpace);
+  }
+
+  assignClasses() {
+    if (this.state.active) {
+      this._root.classList.add(this.constructor.cssClasses.ACTIVE);
+    } else {
+      this._root.classList.remove(this.constructor.cssClasses.ACTIVE);
+    }
+  }
+
+  onClick = () => {
+    this.state.active = !this.state.active;
+    this.assignClasses();
+  };
+
+  onSpace = e => {
+    if (e.code === 'Space') {
```

Comment:
.code lacks IE support https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code ;_;

Reply:
added a check for the actual `keyCode`

Reply:
added a check for the actual `keyCode`

Reply:
lol jk that's deprecated.

_now_ i added a check for `key` which is supported by IE and also not deprecated